torch/distributed/nn/functional.py

```diff
@@ -277,16 +277,16 @@ class _AllGather(Function):
     def forward(ctx, group, tensor):
```

Comment:
Can we also add a non-contiguous input tensor to our tests? e.g. in 
https://github.com/pytorch/pytorch/blob/6a37e0df97777ff1fbd46a27dcd56371f3680ec0/test/distributed/test_c10d_gloo.py#L1092-L1114
or write a new test

Reply:
Here are the instructions for running distributed tests: https://github.com/pytorch/pytorch/blob/master/torch/distributed/CONTRIBUTING.md#running-unit-tests